### PR TITLE
expression: add a failpoint switch to control expr push down in integration test

### PR DESCRIPTION
**Pull request by lonng**
Signed-off-by: Lonng <heng@lonng.org>

### What problem does this PR solve? 

Use the failpoint to control whether to push down an expression in the integration test. Push down all expression if the `failpoint expression` is `all`, otherwise, check whether scalar function's name is contained in the enabled expression list (e.g.`ne,eq,lt`).

Additional: Abstract `IsRequestTypeSupported` to reduce duplicated code in `util/mock/client.go`

### What is changed and how it works?

Add a failpoint to `canFuncBePushed` function.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Comment by lonng** [comment]
@XuHuaiyu @zz-jason PTAL

File: util/mock/client.go [hunk 1]
```diff
@@ -17,53 +17,16 @@ import (
 	"context"
 
 	"github.com/pingcap/tidb/kv"
-	"github.com/pingcap/tipb/go-tipb"
 )
 
 // Client implement kv.Client interface, mocked from "CopClient" defined in
 // "store/tikv/copprocessor.go".
 type Client struct {
+	kv.RequestTypeSupportedChecker
```
Comment (zz-jason) [inline util/mock/client.go]: why make this change?

Comment (lonng) [reply util/mock/client.go]: Remove duplicated code.

Comment (zz-jason) [reply util/mock/client.go]: got it

**Pull request merged**